src/main/scala/com/advancedtelematic/campaigner/db/Campaigns.scala

```diff
@@ -38,30 +41,16 @@ protected [db] class Campaigns(implicit db: Database, ec: ExecutionContext)
       .map(_.headOption)
       .map(_.map(_.processed))
 
-  def completeBatch(
-    ns: Namespace,
-    campaign: CampaignId,
-    group: GroupId,
-    stats: Stats): Future[Unit] =
-    progressGroup(ns, campaign, group, GroupStatus.scheduled, stats)
-
-  def completeGroup(
-    ns: Namespace,
-    campaign: CampaignId,
-    group: GroupId,
-    stats: Stats): Future[Unit] =
-    progressGroup(ns, campaign, group, GroupStatus.launched, stats)
-
-  def cancelCampaign(
-    ns: Namespace,
-    campaign: CampaignId): Future[Unit] = for {
-      _ <- campaignRepo.find(ns, campaign)
-      _ <- cancelTaskRepo.cancel(campaign)
-      _ <- groupStatsRepo.cancel(campaign)
-    } yield ()
+  def completeBatch(campaignId: CampaignId, group: GroupId, stats: Stats): Future[Unit] = db.run {
+    campaignStatusTransition.completeBatch(campaignId, group, stats).transactionally
+  }
 
-  def failedDevices(ns: Namespace, campaign: CampaignId): Future[Set[DeviceId]] = db.run {
-    campaignRepo.findAction(ns, campaign).flatMap { _ =>
+  def completeGroup(campaign: CampaignId, group: GroupId, stats: Stats): Future[Unit] = db.run {
```

Comment:
Sorry, nitpicking here, but maybe name `campaignId` for similarity with the previous method? Must be my OCD kicking in 😆